engine/options.go

```diff
@@ -43,6 +44,11 @@ type (
 	options struct {
 		ds datastore.Batching
 		h  host.Host
+
+		// announceURLs is the list of indexer URLs to send direct HTTP
+		// announce messages to.
+		announceURLs []*url.URL
```

Comment:
Thinking out loud: It's a bit strange that there is straight up URL, multiaddr for DT, _and_ our own flavour of HTTP multiaddr.

Reply:
I did not want to use a multiaddr, because I wanted it to be clear that this is a URL because the protocol is http/https.  I can change this, but it still needs to become a URL at some point.